docker/training/dockerfile.ctr

```diff
@@ -165,27 +234,33 @@ RUN rm -rf /usr/lib/x86_64-linux-gnu/libibverbs.so && \
     ln -s /usr/lib/x86_64-linux-gnu/libibverbs.so.1.14.36.0 /usr/lib/x86_64-linux-gnu/libibverbs.so
 
 # Install hugectr
-RUN if [ "$HUGECTR_DEV_MODE" == "false" ]; then \
-        git clone https://${_CI_JOB_TOKEN}${_HUGECTR_REPO} /hugectr && cd /hugectr && git checkout ${HUGECTR_VER} && \
+ARG HUGECTR_HOME=/usr/local/hugectr
+RUN if [[ "${HUGECTR_DEV_MODE}" == "false" ]]; then \
+        git clone --branch ${HUGECTR_VER} --depth 1 https://${_CI_JOB_TOKEN}${_HUGECTR_REPO} hugectr && \
+        cd hugectr && \
         git submodule update --init --recursive && \
-        mkdir build && cd build && \
+        mkdir build && \
+        cd build && \
         LD_LIBRARY_PATH=/usr/local/cuda/lib64/stubs/:$LD_LIBRARY_PATH && \
-        export PATH=$PATH:/usr/local/cuda-${CUDA_SHORT_VERSION}/compat/ && \
-        cmake -DCMAKE_CXX_COMPILER=/usr/bin/g++ -DCMAKE_C_COMPILER=/usr/bin/gcc -DCMAKE_BUILD_TYPE=Release -DSM="60;61;70;75;80" \
-            -DENABLE_MULTINODES=ON .. && \
-        make -j$(nproc) && make install && \
-        chmod +x /usr/local/hugectr/bin/* && \
-        chmod +x /usr/local/hugectr/lib/* && \
-        cd /hugectr/onnx_converter && \
+        export PATH=$PATH:/usr/local/cuda-${CUDA_SHORT_VERSION}/compat && \
+        if [[ -f "/usr/local/lib/libhdfs.so" ]]; then \
+            cmake -DCMAKE_BUILD_TYPE=Release -DSM="60;61;70;75;80" -DENABLE_MULTINODES=ON -DENABLE_HDFS=ON .. \
+        ; else \
+            cmake -DCMAKE_BUILD_TYPE=Release -DSM="60;61;70;75;80" -DENABLE_MULTINODES=ON .. \
+        ; fi && \
+        make -j$(nproc) && \
+        make install && \
+        chmod +x ${HUGECTR_HOME}/bin/* ${HUGECTR_HOME}/lib/*.so && \
+        cd ../onnx_converter && \
         python setup.py install && \
-        rm -rf /hugectr; \
-    fi
-
-ENV PATH=/usr/local/hugectr/bin:$PATH
-ENV LIBRARY_PATH=/usr/local/hugectr/lib:$LIBRARY_PATH
-ENV LD_LIBRARY_PATH=/usr/local/hugectr/lib:$LD_LIBRARY_PATH
-ENV PYTHONPATH=/usr/local/hugectr/lib:$PYTHONPATH
-ENV PYTHONPATH=/hugectr/onnx_converter:$PYTHONPATH
+        cd ../.. && \
+        rm -rf hugectr \
```

Comment:
I think we need to rebase this with main - we had an issue on the last version where the hugectr tests didn't run as part of the container testing scripts (since the tests weren't included in the /hugectr directory), so we changed to keep the /hugectr directory https://github.com/NVIDIA-Merlin/Merlin/pull/195 so we can run the tests.

Reply:
Done!